### ISSUE TITLE
Select count generating incorrect sql

### ISSUE THREAD

**Issue by Max-Might** [simolus3/moor#529]
Hello,

I have the following code:
```dart
final countFilter = employeeDvirLogs.status.equals(completedStatus) &
        ((employeeDvirLogs.employeeId.equals(employeeId) &
                employeeDvirLogs.clientId.equals(clientId)) |
            employeeDvirLogs.isServerReceived.equals(true)) &
        employeeDvirLogs.unitId.equals(unitId) &
        employeeDvirLogs.isPreTrip.equals(true);

//    final countExpression = countAll(filter: countFilter);
    final countExpression =
        employeeDvirLogs.dvirId.count(distinct: true, filter: countFilter);
    final query = selectOnly(employeeDvirLogs)..addColumns([countExpression]);
    return query.map((row) => row.read(countExpression)).getSingle();
```

which generate the following sql:
```sql
SELECT COUNT(DISTINCT employee_dvir_log.dvir_id) FILTER (WHERE employee_dvir_log.status = ? AND (employee_dvir_log.employee_id = ? AND employee_dvir_log.client_id = ? OR employee_dvir_log.is_server_received = ?) AND employee_dvir_log.unit_id = ? AND employee_dvir_log.is_pre_trip = ?) AS "c0" FROM employee_dvir_log;
```

which produces the following error:
```
Unhandled Exception: DatabaseException(near "(": syntax error (code 1): , while compiling ...
```

So I am not sure if I am using the `count()` API incorrectly (there are not a lot of examples on howto use it) or if the generator gets confused and generates incorrect sql.

**Comment by simolus3** [comment]
Both your code and the generated sql actually looks correct to me, I'm not sure why sqlite wouldn't accept that. 

I think the `FILTER` clause is only available in newer sqlite versions. Are you using `moor_flutter` or `moor_ffi`? Which iOS/Android version are you using?

**Comment by Max-Might** [comment]
I am using `moor_flutter` on Android 8.0.0.

**Comment by simolus3** [comment]
Right. According to [this page](https://developer.android.com/reference/android/database/sqlite/package-summary), Android 8 uses sqlite 3.18. Filter clauses for aggregate versions were added in sqlite 3.30, which has only been released last October.

I've changed the documentation to point out that `filter` requires a very new sqlite version. I think the following query should provide the same results, and it doesn't use `filter` on count:

```dart
final countExpression = countAll();
final query = selectOnly(employeeDvirLogs)..addColumn([count])..where(countFilter);
```
